connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/search/SearchQueryClientImpl.java

```diff
@@ -25,24 +25,31 @@
 
 public class SearchQueryClientImpl implements SearchQueryClient {
 
-  private static final int PAGE_SIZE = 50;
+  private static final int DEFAULT_PAGE_LIMIT = 200;
+  private int limit = DEFAULT_PAGE_LIMIT;
 
   private final CamundaClient camundaClient;
 
   public SearchQueryClientImpl(CamundaClient camundaClient) {
     this.camundaClient = camundaClient;
   }
 
+  SearchQueryClientImpl(CamundaClient camundaClient, int limit) {
+    this.camundaClient = camundaClient;
+    if (limit <= 0) {
+      throw new IllegalArgumentException("Page limit must be greater than zero");
```

Comment:
Why not log a warning "Set page limit was invalid. Must be greater than zero. Falling back to default page limit 200" and proceed?